evm/db/chain.py

```diff
@@ -245,10 +244,11 @@ def clear(self):
         self.db.clear()
 
     def get_state_db(self, state_root, read_only, access_list=None):
-        return AccountStateDB(
-            db=self.db,
-            root_hash=state_root,
-            read_only=read_only,
-            access_list=access_list,
-            backend_class=self.state_backend_class
-        )
+        kwargs = {
+            "db": self.db,
+            "root_hash": state_root,
+            "read_only": read_only,
+        }
```

Comment:
What if instead of constructing the kwargs here you simply made this a passthrough like the following?

```python
def get_state_db(self, state_root, read_only, **extra_kwargs):
    return self.account_state_class(db=self.db, state_root=state_root, read_only=read_only, **extra_kwargs)
```

I'll admit that layers of passthrough `**kwargs` can get really confusing really fast but here it might be clean enough.